File: web3-libs/mod3/playground/aula1/encrypt_wallet.py
from web3 import Account
import json
import getpass


# Cria uma conta nova na wallet e obtém a chave privada
account = Account().create()
private_key = account.key.to_0x_hex()
print("Chave Privada:", private_key)


# Função para solicitar a senha e encriptar a chave privada
def encrypt_private_key():
    password = getpass.getpass("Digite sua senha para encriptar: ")

    keystore = Account().encrypt(private_key, password)
    print("Keystore Encriptado:", json.dumps(keystore, indent=2))

    # Chamar a função para decriptar
    decrypt_keystore(keystore)


# Função para solicitar a senha e decriptar a chave privada
def decrypt_keystore(keystore):
    password = getpass.getpass("Digite sua senha para decriptar: ")

    try:
        decrypted_account = Account().decrypt(keystore, password)
        print("Conta Decriptada:", f'0x{decrypted_account.hex()}')
    except ValueError as error:
        print("Erro ao decriptar a chave:", error)


# Inicia o processo
encrypt_private_key()

File: web3-libs/mod4/playground/aula1/encryptWallet.js
import { ethers } from "ethers";
import readline from "readline";

const PRIVATE_KEY =
  "0xb762c5177d93ae67873060b655b652eecddca5085d918c2de4ca23b1e6be4812";
const wallet = new ethers.Wallet(PRIVATE_KEY);
console.log("PrivateKey:", wallet.privateKey);
console.log("Address:", wallet.address);

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

// Função para solicitar a senha e encriptar a chave privada
const encryptPrivateKey = async () => {
  rl.question("Digite sua senha para encriptar: ", async (password) => {
    rl.close();

    const encryptedWallet = await wallet.encrypt(password);
    console.log("Wallet Encriptado:", encryptedWallet);

    // Chamar a função para decriptar
    decryptWallet(encryptedWallet);
  });
};

// Função para solicitar a senha e decriptar a chave privada
const decryptWallet = (encryptedWallet) => {
  const rlDecrypt = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  rlDecrypt.question("Digite sua senha para decriptar: ", async (password) => {
    rlDecrypt.close();

    try {
      const decryptedWallet = await ethers.Wallet.fromEncryptedJson(
        encryptedWallet,
        password
      );
      console.log("Wallet Decriptado:", decryptedWallet.address);
      console.log("Wallet Decriptado:", decryptedWallet.privateKey);
    } catch (error) {
      console.error("Erro ao decriptar a chave:", error.message);
    }
  });
};

// Inicia o processo
encryptPrivateKey();
